sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/DistinctQueryTests.java

```diff
@@ -0,0 +1,351 @@
+                    if (distinctMap.add(document, outHash)) {
+                        documentsFromWithoutDistinct.add(document);
+                    }
+                }
+            }
+
+
+            CosmosPagedFlux<CosmosItemProperties> queryObservableWithDistinct = createdCollection
+                                                                                    .queryItems(queryWithoutDistinct, options,
+                                                                                                CosmosItemProperties.class);
+
+
+            iterator = queryObservableWithDistinct.byPage().toIterable().iterator();
+
+            while (iterator.hasNext()) {
+                FeedResponse<CosmosItemProperties> next = iterator.next();
+                documentsFromWithDistinct.addAll(next.getResults());
+            }
+
+            assertThat(documentsFromWithDistinct.size()).isEqualTo(documentsFromWithoutDistinct.size());
+        }
+
+    }
+
+    public void bulkInsert() {
+        generateTestData();
+        voidBulkInsertBlocking(createdCollection, docs);
+    }
+
+    public void generateTestData() {
+
+        Random rand = new Random();
+        ObjectMapper mapper = new ObjectMapper();
+        for (int i = 0; i < 40; i++) {
+            Person person = getRandomPerson(rand);
+            try {
+                docs.add(new CosmosItemProperties(mapper.writeValueAsString(person)));
+            } catch (JsonProcessingException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private Pet getRandomPet(Random rand) {
+        String name = getRandomName(rand);
+        int age = getRandomAge(rand);
+        return new Pet(name, age);
+    }
+
+    public Person getRandomPerson(Random rand) {
+        String name = getRandomName(rand);
+        City city = getRandomCity(rand);
+        double income = getRandomIncome(rand);
+        List<Person> people = new ArrayList<Person>();
+        if (rand.nextInt(10) % 10 == 0) {
+            for (int i = 0; i < rand.nextInt(5); i++) {
+                people.add(getRandomPerson(rand));
+            }
+        }
+
+        int age = getRandomAge(rand);
+        Pet pet = getRandomPet(rand);
+        UUID guid = UUID.randomUUID();
+        Person p = new Person(name, city, income, people, age, pet, guid);
+        return p;
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        safeClose(client);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = 3 * SETUP_TIMEOUT)
+    public void beforeClass() throws Exception {
+        client = this.clientBuilder().buildAsyncClient();
+        createdCollection = getSharedMultiPartitionCosmosContainer(client);
+        truncateCollection(createdCollection);
+
+        bulkInsert();
+
+        waitIfNeededForReplicasToCatchUp(clientBuilder());
+    }
+
+    public enum City {
+        NewYork,
+        LosAngeles,
+        Seattle
```

Comment:
java enum code style. take a look at ConsistencyLevel if string value of this matters.

Reply:
changed the test